test/test_sparse_prop.py

```diff
@@ -0,0 +1,97 @@
+# Owner(s): ["module: sparse"]
+
+import torch
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+from typing import Optional, Tuple
+
+
+class SumNet(torch.nn.Module):
+    def __init__(self):
+        super(SumNet, self).__init__()
+        return
+
+    def forward(self, x):
+        return x.sum()
+
+
+class EltwiseNet(torch.nn.Module):
+    def __init__(self):
+        super(EltwiseNet, self).__init__()
+        return
+
+    def forward(self, x):
+        return 2 * torch.sin(-x)
+
+
+# All sparse layouts currently supported in torch.sparse.
+# TODO: make this work for all
+SPARSE_LAYOUTS = [
+    torch.sparse_coo,
+    #    torch.sparse_csr,
+    #    torch.sparse_csc,
+    #    torch.sparse_bsr,
+    #    torch.sparse_bsc,
+]
+
+
+def make_sparse(layout: torch.layout) -> Tuple[torch.Tensor, Optional[Tuple[int, int]]]:
+    dense = torch.ones(4, 8)
+    if layout == torch.sparse_coo:
+        return (dense.to_sparse_coo(), None)
+    if layout == torch.sparse_csr:
+        return (dense.to_sparse_csr(), None)
+    if layout == torch.sparse_csc:
+        return (dense.to_sparse_csc(), None)
+    if layout == torch.sparse_bsr:
+        return (dense.to_sparse_bsr((2, 2)), (2, 2))
+    if layout == torch.sparse_bsc:
+        return (dense.to_sparse_bsc((2, 4)), (2, 4))
+
+
+class TestSparseProp(TestCase):
+    def setUp(self):
+        TestCase.setUp(self)
+
+    def test_sumnet(self):
+        net = SumNet()
+        for sparse_layout in SPARSE_LAYOUTS:
+            sparse_input, blocksize = make_sparse(sparse_layout)
```

Comment:
`make_sparse` gives a sparse tensor with a very simple structure (2D full tensor). For more comprehensive testing with sparse tensors with variable structure, use:
```suggestion
            sparse_input, blocksize = make_sparse(sparse_layout)
            for sparse_input in self.generate_simple_inputs(
                    layout,
                    device='cpu',
                    dtype=torch.float64,
                    index_dtype=torch.int64,
                    # enable_batch=False,
                    # enable_hybrid=False,
            ):
                batch_dim = sparse_input.ndim - sparse_input.sparse_dim() - sparse_input.dense_dim()
                if layout in {torch.sparse_bsr, torch.sparse_bsc}:
                    blocksize = sparse_input.values().shape[batch_dim + 1: batch_dim + 3]
                else:
                    blocksize = None
```

Reply:
That is nice!